src/promptflow-core/promptflow/_utils/flow_utils.py

```diff
@@ -141,20 +148,25 @@ def dump_flow_dag(flow_dag: dict, flow_path: Path):
 
 
 def is_flex_flow(
-    *, file_path: Union[str, Path, None] = None, yaml_dict: Optional[dict] = None, working_dir: Optional[Path] = None
+    *,
+    flow_path: Union[str, Path, PathLike, None] = None,
+    yaml_dict: Optional[dict] = None,
+    working_dir: Union[str, Path, PathLike, None] = None,
+    check_flow_exist=True,
 ):
     """Check if the flow is a flex flow."""
-    if file_path is None and yaml_dict is None:
+    if flow_path is None and yaml_dict is None:
         raise UserErrorException("Either file_path or yaml_dict should be provided.")
-    if file_path is not None and yaml_dict is not None:
+    if flow_path is not None and yaml_dict is not None:
         raise UserErrorException("Only one of file_path and yaml_dict should be provided.")
-    if file_path is not None:
-        file_path = Path(file_path)
-        if working_dir is not None and not file_path.is_absolute():
-            file_path = working_dir / file_path
-        if file_path.suffix.lower() not in [".yaml", ".yml"]:
-            return False
-        yaml_dict = load_yaml(file_path)
+    if flow_path is not None:
+        flow_path, flow_file = resolve_flow_path(flow_path, base_path=working_dir, check_flow_exist=False)
+        file_path = flow_path / flow_file
+        if file_path.is_file() and file_path.suffix.lower() in (".yaml", ".yml"):
+            yaml_dict = load_yaml(file_path)
+        elif not check_flow_exist:
```

Comment:
why we need this parameter?

Reply:
This is to fix a bug
When using pf.run, if there is no flow file, yaml file will be temporarily generated and deleted after creating run.
At this time, run.flow is a non-existent file that cannot be determined by load_yaml and needs to be determined by the run.flow suffix.

![image](https://github.com/microsoft/promptflow/assets/23182548/dde9e0ad-0219-4825-a780-185d6d945785)
![image](https://github.com/microsoft/promptflow/assets/23182548/0d849ca8-c481-479c-9845-0d4282aa6e0c)